src/dune_util/dune_util.mli

```diff
@@ -0,0 +1,40 @@
+module Action = Action
+module Alias_name = Alias_name
+module Build_path_prefix_map = Build_path_prefix_map0
+module Gc = Gc
+module Global_lock = Global_lock
+module Log = Log
+module Persistent = Persistent
+module Report_error = Report_error
+module Stringlike = Stringlike
+
+module type Stringlike = Stringlike_intf.S
+
+open Stdune
+
+val xdg : Xdg.t Lazy.t
+val override_xdg : Xdg.t -> unit
+
+(** The default directory of all caches (build and others), used when
+    environment variables are unset.
+    Set to [$XDG_CACHE_HOME/dune]. *)
+val default_cache_dir : Path.t Lazy.t
+
+val check_absolute : var:string -> path:string -> unit
+
+(** The cache for all the git repositories used in package management.
+    Uses [$DUNE_CACHE_HOME/git-repo] if set, or
+    [default_cache_dir/git-repo] otherwise. *)
+val git_repo_cache : Path.t Lazy.t
+
+(** The cache for expensive git revision operations.
+    Uses [$DUNE_CACHE_HOME/rev_store] if set, or
+    [default_cache_dir/rev_store] otherwise. *)
+val revision_store_cache : Path.t Lazy.t
+
+(** The cache for the toolchains, containing non-relocatable packages.
+    Uses [$DUNE_CACHE_HOME/toolchains] if set, or
+    [default_cache_dir/toolchains] otherwise. *)
+val toolchains_cache : Path.t Lazy.t
```

Comment:
If we move these into the respective components where they are used, I think we also don't need to document the concrete locations, which seems like a plus to me, since these should be private to the locations that use them, I think?